add a doc for node expansion in English (#204)
hwameistor/hwameistor: docs/docs/quick_start/advanced_features/node_expansion.md

---

# Node Expansion

> It's one of common requirements for a storage system to expand the capacity by adding a new storage node. HwameiStor provides this feature by the following simple procedure.

Reviewer: ```suggestion
A storage system is ususally expected to expand its capacity by adding a new storage node. In HwameiStor, it can be done with the following steps.
```

Author: better :) fixed

> It's one of common requirements for a storage system to expand the capacity by adding a new storage node. HwameiStor provides this feature by the following simple procedure.

Author: fixed

## Steps

> ### 1. Prepare a storage node (e.g. k8s-worker-4, /dev/sdb, SSD disk)

Reviewer 2: ```suggestion
### 1. Prepare a new storage node

Firstly check nodes in the current Kubernetes cluster:

```console
$ kubectl get no  
NAME           STATUS   ROLES            AGE     VERSION
k8s-master-1   Ready    master           96d     v1.24.3-2+63243a96d1c393
k8s-worker-1   Ready    piraeus,worker   6d23h   v1.24.3-2+63243a96d1c393
k8s-worker-2   Ready    piraeus,worker   6d22h   v1.24.3-2+63243a96d1c393
k8s-worker-3   Ready    piraeus,worker   50d     v1.24.3-2+63243a96d1c393
```
```

Reviewer: maybe add a comma after "firstly"?

Author: fixed

> Add a node into the Kubernetes cluster, or select a Kubernetes node. And ensure the node has all the required items which are described in [Prerequisites](../install/prereq.md).

Reviewer 2: ```suggestion
Add a new node into the Kubernetes cluster, or select a Kubernetes node. And ensure the node has all the required items which are described in [Prerequisites](../install/prereq.md).

The new node information is as follows:

- name: k8s-worker-4
- devPath: /dev/sdb
- diskType: SSD disk
```

Author: fixed

> Add a node into the Kubernetes cluster, or select a Kubernetes node. And ensure the node has all the required items which are described in [Prerequisites](../install/prereq.md).

Reviewer: ```suggestion
Add the node into the Kubernetes cluster, or select a Kubernetes node. The node should have all the required items described in [Prerequisites](../install/prereq.md).
```

Author: fixed

> After the new node is already in the Kubernetes cluster, make sure the following HwameiStor pods are already running on this node.

Reviewer 2: ```suggestion
After the new node is added in the Kubernetes cluster, make sure the following HwameiStor pods are running on this node.
```

Reviewer: "added into" sounds better to me.

Author: fixed

```console
$ kubectl -n hwameistor get pod -o wide | grep k8s-worker-4

> [root@demo-dev-master-01 ~]# k -n hwameistor get pod -o wide | grep demo-dev-worker-02

Reviewer 2: ```suggestion
```

Author: fixed

hwameistor-local-disk-manager-c86g5     2/2     Running   0     19h   10.6.182.105      k8s-worker-4   <none>  <none>
hwameistor-local-storage-s4zbw          2/2     Running   0     19h   192.168.140.82    k8s-worker-4   <none>  <none>

> # check for the LSN which is for the metadata of the node's storage

Reviewer 2: ```suggestion
# check the LSN which is for the metadata of the node's storage
```

Author: fixed

> # check for the LSN which is for the metadata of the node's storage

Reviewer: ```suggestion
# check LocalStorageNode to get metadata of the storage node.
```
The original sentence is not easy to understand. Try to rewrite as the above.

Author: fixed

$ kubectl get localstoragenode k8s-worker-4
NAME                 IP           ZONE      REGION    STATUS   AGE
k8s-worker-4   10.6.182.103       default   default   Ready    8d
```

### 2. Add the storage node into HwameiStor

Add the storage label into the node by adding a LocalStorageClaim CR as below:

```console
$ kubectl apply -f - <<EOF
apiVersion: hwameistor.io/v1alpha1
kind: LocalDiskClaim
metadata:
  name: k8s-worker-4
spec:
  nodeName: k8s-worker-4
  description:
    diskType: SSD
EOF
```

### 3. Post check

> Finally, check if the node has the storage pool setup by following:

Reviewer 2: ```suggestion
Finally, check if the node has set up the storage pool by checking:
```

Reviewer: how about "by checking this file"?

Author: fixed
